service/controller/clusterapi/v30/resource/asgstatus/create.go

```diff
@@ -63,13 +63,19 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
 			return nil
 		}
-		if len(o.Reservations[0].Instances) == 0 {
+		nonTerminatedInstances := o.Reservations[0].Instances[:0]
```

Comment:
A couple of thoughts. What about `runningInstances` rather than `nonTerminatedInstances`?

Also its personal taste but I'm not sure about the slice magic. What type is this slice? I think defining it as an empty slice might be more readable.

Reply:
Please check https://github.com/giantswarm/aws-operator/blob/9837654f41a9f1be2acdbba6ea5b5d075b54e4ad/service/controller/clusterapi/v31/resource/tccp/resource.go#L114-L179.

Reply:
Thanks @xh3b4sd! Applied server side instead of client side filtering. Less code, no naming issues. PTAL

Reply:
Merged to be able to continue with next and most important cherry pick - k8s+helm updates PR. If you still have comments feel free to leave them, can incorporate in a followup PR.

Reply:
All good from my side. Using the filters is very nice!